tools/c7n_azure/c7n_azure/function_package.py

```diff
@@ -226,11 +222,7 @@ def publish(self, deployment_creds):
 
         self.log.info("Publishing Function package from %s" % self.pkg.path)
 
-        # Windows requires TEMPORARY flag if you want to open files created by tempfile library
-        if os.name == 'nt':
-            zip_file = open(self.pkg.path, 'rb', opener=FunctionPackage._temporary_opener).read()
-        else:
-            zip_file = open(self.pkg.path, 'rb').read()
+        zip_file = self.pkg.open()
```

Comment:
self.pkg.open()`.read()`

zip_file is file content, not a stream

Reply:
I have updated and will be included in my next commit.